dns.tf

```diff
@@ -7,3 +7,53 @@ data "azurerm_resource_group" "proddns_jenkinsio" {
 data "azurerm_dns_zone" "jenkinsio" {
   name = "jenkins.io"
 }
+
+resource "azurerm_dns_zone" "child_zones" {
+  for_each = local.lets_encrypt_dns_challenged_domains
+
+  name = each.key
+  # Use the same resource group for all DNS zones
+  resource_group_name = data.azurerm_resource_group.proddns_jenkinsio.name
+
+  tags = local.default_tags
+}
+
+# create DNS record of type NS for child-zone in the parent zone (to allow propagation of DNS records)
+resource "azurerm_dns_ns_record" "child_zone_ns_records" {
+  for_each = local.lets_encrypt_dns_challenged_domains
+
+  name                = trimsuffix(each.key, ".jenkins.io") # only the flat name not the fqdn
+  zone_name           = data.azurerm_dns_zone.jenkinsio.name
+  resource_group_name = data.azurerm_resource_group.proddns_jenkinsio.name
+  ttl                 = 60
+
+  records = azurerm_dns_zone.child_zones[each.key].name_servers
+}
+
+resource "azuread_application" "letsencrypt_dns_challenges" {
```

Comment:
It's generally recommended to set the service principal that created the application as an owner to help you identify where it was created from / who to contact, (it's in the example at: https://registry.terraform.io/providers/hashicorp/azuread/latest/docs/resources/application)

you could probably also configure the apps home page to point to this repo

Reply:
Changes applied